Focus jumps when I have two tables on the same screen and scroll the above table

The version "1.0.0-beta.5" has the following bug:
When I have 2 tables on the same page, focus the bottom one, scroll the top one and then try to write something on the bottom table (which still has the focus on), the focus jumps all over the place.

Like this:
![bug](https://user-images.githubusercontent.com/36645390/49936102-f27e1c00-feca-11e8-853a-9e9f10122b75.gif)

## Your Environment
* Version used: 1.0.0-beta.5
* Browser Name and version: Chrome latest
* Operating System and version (desktop or mobile): mac desktop
* Link to your project: https://github.com/Powerade/fixed-data-table-2-bugs/tree/focus-bug-double-table

Hey man, thanks for the quick response.
I'm trying to run that branch with:
`"fixed-data-table-2": "git+https://github.com/schrodinger/fixed-data-table-2.git#multi_table_fix",`

But I'm getting the following error:
./node_modules/fixed-data-table-2/main.js
Module not found: Can't resolve './internal/FixedDataTableRoot' in '/Users/<user>/fixed-data-table-2-bugs/node_modules/fixed-data-table-2'

Amd I doing something wrong here?

--- a/src/reducers/index.js
+++ b/src/reducers/index.js
@@ -21,11 +21,18 @@ import convertColumnElementsToData from 'convertColumnElementsToData';
 import pick from 'lodash/pick';
 import shallowEqual from 'shallowEqual';
 
+/**
+ * Returns the default initial state for the redux store.
+ * This must be a brand new, independent object for each table instance
+ * or issues may occur due to multiple tables sharing data.
+ *
+ * @return {!Object}
+ */
 function getInitialState() {
-  /**
-   * Input state set from props
-   */
-  const DEFAULT_INPUT_STATE = {
+  return {
+    /*
+     * Input state set from props
+     */
     columnProps: [],
     columnGroupProps: [],
     elementTemplates: {
@@ -60,13 +67,11 @@ function getInitialState() {
       useMaxHeight: false,
       width: 0,
     },
-  };
 
-  /**
-   * Output state passed as props to the the rendered FixedDataTable
-   * NOTE (jordan) rows may contain undefineds if we don't need all the buffer positions
-   */
-  const DEFAULT_OUTPUT_STATE = {
+    /*
+     * Output state passed as props to the the rendered FixedDataTable
+     * NOTE (jordan) rows may contain undefineds if we don't need all the buffer positions
+     */
     columnReorderingData: {},
     columnResizingData: {},
     firstRowIndex: 0,
@@ -80,24 +85,17 @@ function getInitialState() {
     scrollContentHeight: 0,
     scrollX: 0,
     scrollY: 0,
-  };
 
-  /**
-   * Internal state only used by this file
-   * NOTE (jordan) internal state is altered in place
-   * so don't trust it for redux history or immutability checks
-   * TODO (jordan) investigate if we want to move this to local or scoped state
-   */
-  const DEFAULT_INTERNAL_STATE = {
+    /*
+     * Internal state only used by this file
+     * NOTE (jordan) internal state is altered in place
+     * so don't trust it for redux history or immutability checks
+     * TODO (jordan) investigate if we want to move this to local or scoped state
+     */
     bufferSet: new IntegerBufferSet(),
     storedHeights: [],
     rowOffsets: null, // PrefixIntervalTree
   };
-
-  return Object.assign({},
-    DEFAULT_INPUT_STATE,
-    DEFAULT_OUTPUT_STATE,
-    DEFAULT_INTERNAL_STATE);
 }
 
 function reducers(state = getInitialState(), action) {

--- a/src/reducers/index.js
+++ b/src/reducers/index.js
@@ -21,84 +21,86 @@ import convertColumnElementsToData from 'convertColumnElementsToData';
 import pick from 'lodash/pick';
 import shallowEqual from 'shallowEqual';
 
-/**
- * Input state set from props
- */
-const DEFAULT_INPUT_STATE = {
-  columnProps: [],
-  columnGroupProps: [],
-  elementTemplates: {
-    cell: [],
-    footer: [],
-    groupHeader: [],
-    header: [],
-  },
-  elementHeights: {
-    footerHeight: 0,
-    groupHeaderHeight: 0,
-    headerHeight: 0,
-  },
-  rowSettings: {
-    bufferRowCount: undefined,
-    rowHeight: 0,
-    rowHeightGetter: () => 0,
-    rowsCount: 0,
-    subRowHeight: 0,
-    subRowHeightGetter: () => 0,
-  },
-  scrollFlags: {
-    overflowX: 'auto',
-    overflowY: 'auto',
-    showScrollbarX: true,
-    showScrollbarY: true,
-  },
-  tableSize: {
-    height: undefined,
-    maxHeight: 0,
-    ownerHeight: undefined,
-    useMaxHeight: false,
-    width: 0,
-  },
-};
+function getInitialState() {
+  /**
+   * Input state set from props
+   */
+  const DEFAULT_INPUT_STATE = {
+    columnProps: [],
+    columnGroupProps: [],
+    elementTemplates: {
+      cell: [],
+      footer: [],
+      groupHeader: [],
+      header: [],
+    },
+    elementHeights: {
+      footerHeight: 0,
+      groupHeaderHeight: 0,
+      headerHeight: 0,
+    },
+    rowSettings: {
+      bufferRowCount: undefined,
+      rowHeight: 0,
+      rowHeightGetter: () => 0,
+      rowsCount: 0,
+      subRowHeight: 0,
+      subRowHeightGetter: () => 0,
+    },
+    scrollFlags: {
+      overflowX: 'auto',
+      overflowY: 'auto',
+      showScrollbarX: true,
+      showScrollbarY: true,
+    },
+    tableSize: {
+      height: undefined,
+      maxHeight: 0,
+      ownerHeight: undefined,
+      useMaxHeight: false,
+      width: 0,
+    },
+  };
 
-/**
- * Output state passed as props to the the rendered FixedDataTable
- * NOTE (jordan) rows may contain undefineds if we don't need all the buffer positions
- */
-const DEFAULT_OUTPUT_STATE = {
-  columnReorderingData: {},
-  columnResizingData: {},
-  firstRowIndex: 0,
-  firstRowOffset: 0,
-  isColumnReordering: false,
-  isColumnResizing: false,
-  maxScrollX: 0,
-  maxScrollY: 0,
-  rowHeights: {},
-  rows: [], // rowsToRender
-  scrollContentHeight: 0,
-  scrollX: 0,
-  scrollY: 0,
-};
+  /**
+   * Output state passed as props to the the rendered FixedDataTable
+   * NOTE (jordan) rows may contain undefineds if we don't need all the buffer positions
+   */
+  const DEFAULT_OUTPUT_STATE = {
+    columnReorderingData: {},
+    columnResizingData: {},
+    firstRowIndex: 0,
+    firstRowOffset: 0,
+    isColumnReordering: false,
+    isColumnResizing: false,
+    maxScrollX: 0,
+    maxScrollY: 0,
+    rowHeights: {},
+    rows: [], // rowsToRender
+    scrollContentHeight: 0,
+    scrollX: 0,
+    scrollY: 0,
+  };
 
-/**
- * Internal state only used by this file
- * NOTE (jordan) internal state is altered in place
- * so don't trust it for redux history or immutability checks
- * TODO (jordan) investigate if we want to move this to local or scoped state
- */
-const DEFAULT_INTERNAL_STATE = {
-  bufferSet: new IntegerBufferSet(),
-  storedHeights: [],
-  rowOffsets: null, // PrefixIntervalTree
-};
+  /**
+   * Internal state only used by this file
+   * NOTE (jordan) internal state is altered in place
+   * so don't trust it for redux history or immutability checks
+   * TODO (jordan) investigate if we want to move this to local or scoped state
+   */
+  const DEFAULT_INTERNAL_STATE = {
+    bufferSet: new IntegerBufferSet(),
+    storedHeights: [],
+    rowOffsets: null, // PrefixIntervalTree
+  };
 
-const DEFAULT_STATE = Object.assign({},
-  DEFAULT_INPUT_STATE,
-  DEFAULT_OUTPUT_STATE,
-  DEFAULT_INTERNAL_STATE);
+  return Object.assign({},
+    DEFAULT_INPUT_STATE,
+    DEFAULT_OUTPUT_STATE,
+    DEFAULT_INTERNAL_STATE);
+}
 
-function reducers(state = DEFAULT_STATE, action) {
+function reducers(state = getInitialState(), action) {
   switch (action.type) {
     case ActionTypes.INITIALIZE: {
       const { props } = action;

